pinctrl/gpio.go

```diff
@@ -31,20 +34,32 @@ type gpioPin struct {
 	pwmFreqHz       uint
 	pwmDutyCyclePct float64
 
-	mu        sync.Mutex
-	cancelCtx context.Context
-	logger    logging.Logger
+	mu     sync.Mutex
+	logger logging.Logger
+}
 
-	swPwmCancel func()
+// CreateGpioPin creates a gpio pin.
+func CreateGpioPin(mapping gl.GPIOBoardMapping, gpioPinsPage *mmap.MMap, logger logging.Logger) *GPIOPin {
+	pin := GPIOPin{
+		devicePath: mapping.GPIOChipDev,
+		offset:     uint32(mapping.GPIO),
+		logger:     logger,
+	}
+	if mapping.HWPWMSupported {
+		pin.hwPwm = newPwmDevice(mapping.PWMSysFsDir, mapping.PWMID, logger, gpioPinsPage)
```

Comment:
You know, I think the generic Linux board mapping could be gotten rid of setting this code fully - the pull-ups and pull-downs do not use it, and neither do the hw pwm setting. The only things left are the gpio set/get and the software PWM logic. kay to think about later.

Reply:
decided to look at this more to try learning whats going on, as I am not too familiar with the code. each gpio/di/pwm is initialized using information from `mapping`, namely the device path and offset fields. 

at a glance I am not sure how i would untangle `mapping` from the gpio/di/pwm logic. It looks like `mapping` is just a bunch of board specific fields we defined somewhere else in rdk, but I don't necessarily think theres a better solution than just having some struct store all of this information. Its also very possible the info in `mapping` is redundant with other information in this code too.